compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -824,16 +813,20 @@ object TreeChecker {
         && !isPrimaryConstructorReturn
         && !pt.isInstanceOf[FunOrPolyProto]
       then
-        assert(tree.tpe <:< pt, {
-          val mismatch = TypeMismatch(tree.tpe, pt, Some(tree))
-          i"""|Type Mismatch:
-              |${mismatch.message}
-              |tree = $tree ${tree.className}""".stripMargin
-        })
+        checkType(tree.tpe, pt, tree, "adapt")
       tree
     }
 
     override def simplify(tree: Tree, pt: Type, locked: TypeVars)(using Context): tree.type = tree
+
+    private def checkType(tp1: Type, tp2: Type, tree: untpd.Tree, step: String)(using Context) =
+      // Accept NoType <:< NoType as true
+      assert((tp1 eq tp2) || (tp1 <:< tp2), {
+        val mismatch = TypeMismatch(tp1, tp2, None)
```

Comment:
Shouldn't it be ?
```suggestion
        val mismatch = TypeMismatch(tp1, tp2, Some(tree))
```

Reply:
No, because I'm adding .className to the tree ouput.